### PR TITLE
Refactor a small function

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

**Comment by timgraham** [comment]
Did you consider all `while True` loops as well? I think all changes (e.g. including https://github.com/django/django/pull/9523) could be done in the same commit.

**Comment by srinivasreddy** [comment]
I will be doing it.

**Comment by srinivasreddy** [comment]
@timgraham  Done. Rest of them too complex to take up now. But can be done later.

File: django/contrib/sessions/backends/base.py [hunk 1]
```diff
@@ -148,10 +148,9 @@ def is_empty(self):
 
     def _get_new_session_key(self):
         "Return session key that isn't being used."
-        while True:
+        session_key = get_random_string(32, VALID_KEY_CHARS)
```
Comment (timgraham) [inline django/contrib/sessions/backends/base.py]: I'm not sure if refactors like this are useful considering that it makes the code less DRY (`get_random_string(32, VALID_KEY_CHARS)` is repeated). I'd be inclined to leave it as is.

Comment (srinivasreddy) [reply django/contrib/sessions/backends/base.py]: Couple of them below looks similar too. I am leaving  them too. Thanks.

File: django/utils/datetime_safe.py [hunk 2]
```diff
@@ -60,13 +60,10 @@ def new_datetime(d):
...
     # Also finds overlaps
     sites = []
-    i = 0
-    while 1:
-        j = text.find(substr, i)
-        if j == -1:
-            break
-        sites.append(j)
-        i = j + 1
+    i = text.find(substr, 0)
```
Comment (timgraham) [inline django/utils/datetime_safe.py]: 0 is uneeded

File: django/utils/datetime_safe.py [hunk 3]
```diff
@@ -60,13 +60,10 @@ def new_datetime(d):
...
+    while i > -1:
+        sites.append(i)
+        i = text.find(substr, i + 1)
```
Comment (timgraham) [inline django/utils/datetime_safe.py]: While this duplicates, it seems like an improvement in readability since `j` is eliminated.

**Comment by srinivasreddy** [comment]
Looks like `while True` can never be avoided because python disallows `while (x = somecond/function):` technique.

**Pull request merged**